testapi.pm

```diff
@@ -1549,6 +1571,72 @@ sub mouse_hide(;$) {
     query_isotovideo('backend_mouse_hide', {offset => $border_offset});
 }
 
+=head2 mouse_drag
+  mouse_drag([$startpoint, $endpoint, $startx, $starty, $endx, $endy, $button, $timeout]);
+
+Click mouse C<$button>, C<'left'> or C<'right'>, at a given location, hold the button and drag 
+the mouse to another location where the button is released. You can set the C<$startpoint> 
+and C<$endpoint> by passing the name of the needle tag, i.e. the mouse drag happens between
+the two needle areas. Alternatively, you can set all the coordinates explicitly with C<$startx>,
+C<$starty>, C<$endx>, and C<$endy>. You can also set one point using a needle and another one
+using coordinates.  If both the coordinates and the needle are provided, the coordinates 
+will be used to set up the locations and the needle location will be overridden.
+
+=cut
+
+sub mouse_drag {
+    my %args = @_;
+    my $startx =
+      my $starty =
+      my $endx =
+      my $endy = 0;
+    # If full coordinates are provided, work with them as a priority,
+    if (defined $args{startx} and defined $args{starty}) {
+        $startx = $args{startx};
+        $starty = $args{starty};
+    }
+    # If the coordinates were not complete, use the needle as a fallback solution.
+    elsif (defined $args{startpoint}) {
+        my $startmatch = $args{startpoint};
+        # Check that the needle exists.
+        my $start_matched_needle = assert_screen($startmatch, $args{timeout});
+        # Calculate the click point from the area defined by the needle (take the center of it)
+        ($startx, $starty) = calculate_clickpoint($start_matched_needle);
+    }
+    # If neither coordinates nor a needle is provided, report an error and quit.
+    else {
+        die "The starting point of the drag was not correctly provided. Either provide the 'startx' and 'starty' coordinates, or a needle marking the starting point.";
+    }
+
+    # Repeat the same for endpoint coordinates or needles.
+    if (defined $args{endx} and defined $args{endy}) {
+        $endx = $args{endx};
+        $endy = $args{endy};
+    }
+    elsif (defined $args{endpoint}) {
+        my $endmatch = $args{endpoint};
+        my $end_matched_needle = assert_screen($endmatch, $args{timeout});
+        ($endx, $endy) = calculate_clickpoint($end_matched_needle);
+    }
+    else {
+        die "The ending point of the drag was not correctly provided. Either provide the 'endx' and 'endy' coordinates, or a needle marking the starting point.";
+    }
+    # Get the button variable. If no button has been provided, assume the "left" button.
+    my $button = $args{button} //= "left";
```

Comment:
Is the assignment to `$args{button}` intended? I'd removed it because it looks like it isn't used in subsequent code anyways.

Reply:
I think it is used for the `query_isotovideo` calls on lines 1631 and 1633 at least.

Reply:
But you're just using `$button` there. Note that I'm talking about the assignment of `$args{button}`. I suggest to replace `//=` with just `//`.